pkg/api/v1beta2/dynakube/dynakube_conversion.go

```diff
@@ -0,0 +1,160 @@
+
+	if hostMonitoring := src.Spec.OneAgent.HostMonitoring; hostMonitoring != nil {
+		if hostMonitoring.SecCompProfile != "" {
+			dst.Annotations[dynakube.AnnotationFeatureOneAgentSecCompProfile] = hostMonitoring.SecCompProfile
+		}
+	}
+
+	if src.Spec.OneAgent.CloudNativeFullStack != nil {
+		if !isLabelSelectorEmpty(src.Spec.OneAgent.CloudNativeFullStack.NamespaceSelector) {
+			matchExpressions := src.Spec.OneAgent.CloudNativeFullStack.NamespaceSelector.MatchExpressions
+			matchLabels := src.Spec.OneAgent.CloudNativeFullStack.NamespaceSelector.MatchLabels
+
+			dst.Spec.NamespaceSelector = v1.LabelSelector{
+				MatchExpressions: matchExpressions,
+				MatchLabels:      matchLabels,
+			}
+		}
+	} else if src.Spec.OneAgent.ApplicationMonitoring != nil {
+		if !isLabelSelectorEmpty(src.Spec.OneAgent.ApplicationMonitoring.NamespaceSelector) {
+			matchExpressions := src.Spec.OneAgent.ApplicationMonitoring.NamespaceSelector.MatchExpressions
+			matchLabels := src.Spec.OneAgent.ApplicationMonitoring.NamespaceSelector.MatchLabels
+
+			dst.Spec.NamespaceSelector = v1.LabelSelector{
+				MatchExpressions: matchExpressions,
+				MatchLabels:      matchLabels,
+			}
+		}
+	}
+
+	dst.Status.OneAgent.Instances = map[string]dynakube.OneAgentInstance{}
+
+	for key, value := range src.Status.OneAgent.Instances {
+		tmp := dynakube.OneAgentInstance{
+			PodName:   value.PodName,
+			IPAddress: value.IPAddress,
+		}
+		dst.Status.OneAgent.Instances[key] = tmp
+	}
+
+	dst.Status.OneAgent.Version = src.Status.OneAgent.Version
+
+	dst.Status.Phase = src.Status.Phase
+	dst.Status.UpdatedTimestamp = src.Status.UpdatedTimestamp
+
+	return nil
+}
+
+func isLabelSelectorEmpty(selector v1.LabelSelector) bool {
+	return len(selector.MatchExpressions) == 0 && len(selector.MatchLabels) == 0
+}
+
+// ConvertFrom converts v1beta1 to v1beta2.
+func (dst *DynaKube) ConvertFrom(srcRaw conversion.Hub) error {
+	src := srcRaw.(*dynakube.DynaKube)
+	dst.ObjectMeta = src.ObjectMeta
+
+	// DynakubeSpec
+	dst.Spec.APIURL = src.Spec.APIURL
+	dst.Spec.Tokens = src.Spec.Tokens
+	dst.Spec.CustomPullSecret = src.Spec.CustomPullSecret
+	dst.Spec.SkipCertCheck = src.Spec.SkipCertCheck
+	dst.Spec.Proxy = (*DynaKubeProxy)(src.Spec.Proxy)
+	dst.Spec.TrustedCAs = src.Spec.TrustedCAs
+	dst.Spec.NetworkZone = src.Spec.NetworkZone
+	dst.Spec.EnableIstio = src.Spec.EnableIstio
+
+	if src.Annotations == nil {
+		src.Annotations = map[string]string{}
+	}
+
+	if src.Annotations[dynakube.AnnotationFeatureMetadataEnrichment] == "true" {
+		dst.Spec.MetaDataEnrichment = MetaDataEnrichment{
+			Enabled: false,
+		}
+	}
+
+	src.Annotations[dynakube.AnnotationFeatureApiRequestThreshold] = strconv.FormatInt(int64(dst.Spec.DynatraceApiRequestThreshold), 10)
+
+	if src.Annotations[dynakube.AnnotationFeatureOneAgentSecCompProfile] != "" {
+		secCompProfile := src.Annotations[dynakube.AnnotationFeatureOneAgentSecCompProfile]
+		dst.Spec.OneAgent.HostMonitoring.SecCompProfile = secCompProfile
+	}
```

Comment:
```suggestion
	if src.Annotations[dynakube.AnnotationFeatureOneAgentSecCompProfile] != "" {
		secCompProfile := src.Annotations[dynakube.AnnotationFeatureOneAgentSecCompProfile]
		dst.Spec.OneAgent.HostMonitoring.SecCompProfile = secCompProfile
		delete(dst.Annotations, dynakube.AnnotationFeatureOneAgentSecCompProfile)
	}
```

should remove the "moved" feature-flag, so they don't get out of sync (so its nice and clean)

Reply:
should be done in https://github.com/Dynatrace/dynatrace-operator/pull/3092/commits/71219b23d3ca62cab9944fc42f986e38da78c886